src/DataFrames.jl

```diff
@@ -128,4 +128,18 @@ for (dir, filename) in [
     include(joinpath(dir, filename))
 end
 
+# Import  "StringEncodings" if available and set up decoder
+_decoder=nothing
+ENCODING_PACKAGE_READY = false
+function __init__()
+    try Pkg.installed("StringEncodings")
```

Comment:
Thank you @yuyichao to trying to clarify. I meant that the `StringEncodings` will be inside `~/.julia/`, not in the  default `LOAD_PATH`  array, in my system it is something like

```julia
julia> LOAD_PATH
2-element Array{Any,1}:
 "/home/someusr/temp/juliaExpr/julia0.6/usr/local/share/julia/site/v0.6"
 "/home/someusr/temp/juliaExpr/julia0.6/usr/share/julia/site/v0.6"      
```
I might be missing something here as to the meaning of `LOAD_PATH` . 

> If optional dependencies are changed, doing it at compile time will require a manual recompilation only when an optional dependency is added, 

I agree. 
>while doing it in __init__ will break every packages depening on this one, since you suddenly starts to include dependencies that the precompile system did not see before.

Just to make sure, you are saying the packages the require `DataFrames` will broken. right? Are you saying a precompiled package with dependencies will not be calling the __init__ of its dependent packages when loading.  

Or, more concretely, if a package `X` depends on  `DataFrames` that was initially installed without `StringEncodings`. After installing `StringEncodings`, package `X` will still not be able to use feature in `DataFrames` that `StringEncodings` enable. Right?